actors/systemsettings/src/main/java/org/sunbird/systemsettings/actors/SystemSettingsActor.java

```
@@ -0,0 +1,119 @@
+import org.sunbird.systemsettings.service.SystemSettingService;
+import org.sunbird.systemsettings.service.impl.SystemSettingServiceImpl;
+
+@ActorConfig(
+  tasks = {"getSystemSettingById", "getAllSystemSettings", "updateSystemSettingById"},
+  asyncTasks = {}
+)
+public class SystemSettingsActor extends BaseActor {
+  private final ObjectMapper mapper = new ObjectMapper();
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private SystemSetting systemSetting;
+  private final SystemSettingService systemSettingService =
+      new SystemSettingServiceImpl(cassandraOperation);
+  private final Integer WRITE_SETTINGS_RETRIES_ALLOWED = 1;
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    Util.initializeContext(request, TelemetryEnvKey.SYSTEM_SETTINGS);
+    ExecutionContext.setRequestId(request.getRequestId());
+    if (request.getOperation().equalsIgnoreCase(ActorOperations.GET_SYSTEM_SETTING.getValue())) {
+      getSystemSettingById(request);
+    } else if (request
+        .getOperation()
+        .equalsIgnoreCase(ActorOperations.GET_ALL_SYSTEM_SETTINGS.getValue())) {
+      getAllSystemSettings();
+    } else if (request
+        .getOperation()
+        .equalsIgnoreCase(ActorOperations.SET_SYSTEM_SETTING.getValue())) {
+      updateSystemSettingById(request);
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+  /** @param actorMessage Instance of Request class contains the Setting id data */
+  @SuppressWarnings("unchecked")
+  private void getSystemSettingById(Request actorMessage) {
+    ProjectLogger.log("getSystemSettingById method call started", LoggerEnum.DEBUG.name());
+    try {
+      Map<String, Object> req = (Map<String, Object>) actorMessage.getRequest();
+      SystemSetting initSetting = systemSettingService.readSetting((String) req.get(JsonKey.ID));
+      Response response = new Response();
```

If setting value is null, we should raise an error.